pkg/server/volume.go

```diff
@@ -47,6 +47,14 @@ func mountVolume(volume string) error {
 	return mounter.FormatAndMount(devicePath, mountPath, "ext4", nil)
 }
 
+func setPermissions(volume string, mode os.FileMode) error {
+	if !checkMountValid(volume) {
+		return fmt.Errorf("cannot set permissions %v for volume %v invalid mount point", mode, volume)
+	}
+	mountPath := filepath.Join(exportPath, volume)
+	return os.Chmod(mountPath, mode)
```

Comment:
Just in case, should we do recursive `chmod`? Or do we need to introduce a policy similar to `fsGroupChangePolicy`?

Reply:
Since at the moment, we have no way for the user to specify the export config (acl, permissions, squash, etc)
It makes more sense to only make the root folder world writeable, since the subfolders/files should retain the ownership with which they where created. 

So for example if the users workload specifies user id/group 1000, the files will be created with id/group 1000 as owner and have the appropriate non world writeable permissions.

A future idea is to make the per volume export config a configmap, that can be user specified, this way they can setup client acls for the nfs export and specify whether they want squash or any other behaviors.

If the user wants to transfer ownership to a different id, then they can start a root container (init) to transfer/chmod the files/folders recursively. But I think this is actually not what most people would want to do.

Instead they would assume that the files/folders retain ownership with which they have been created. By not doing the recursive chown we also don't run into the long startup time problem with volumes with lots of files.

Reply:
OK. Or the users need to make sure all related workloads have the correct permission. Otherwise, they need to launch the init container to do recursive permission change.

Reply:
This does not work, I think you should call syscall.Umask(0) to configure the system's Umask before modifying the permissions, and then use os.chmod(0777), otherwise the generated permissions will be RWX r-x r-x.